test/integration/consume_test.exs

```diff
@@ -0,0 +1,242 @@
+               fetch_config: [
+                 max_bytes: 10_240
+               ]
+             ]},
+          concurrency: 3
+        ],
+        processors: [
+          default: [
+            concurrency: 3
+          ]
+        ],
+        batchers: [
+          default: [
+            batch_size: 20,
+            batch_timeout: 50,
+            concurrency: 4
+          ]
+        ]
+      )
+    end
+
+    def handle_message(_, message, %{caller_pid: caller_pid}) do
+      if message.data in Config.last_messages() do
+        send(caller_pid, {:last_message, message.metadata.partition, message.data})
+      end
+
+      message
+    end
+
+    def handle_batch(_, messages, _info, %{messages_agent: messages_agent}) do
+      Agent.update(messages_agent, fn list -> list ++ messages end)
+      last_message = List.last(messages)
+      last_offset = last_message.metadata.offset
+      partition = last_message.metadata.partition
+
+      IO.puts(
+        "Batch handled with #{length(messages)} messages. " <>
+          "Partition: #{partition} Last offset: #{last_offset}"
+      )
+
+      messages
+    end
+  end
+
+  setup_all do
+    topic = "test"
+    hosts = [localhost: 9092]
+
+    reset_topic(topic)
+
+    {broadway_pid, messages_agent} = start_broadway()
+
+    # Let's wait a bit to make sure broadway has started
+    Process.sleep(3000)
+
+    send_messages(Config.n_messages(), hosts, topic)
+
+    [last_message_2, last_message_0, last_message_1] = Config.last_messages()
+
+    receive do
+      {:last_message, 2, ^last_message_2} ->
+        IO.puts("Got last message from partition 2")
+    end
+
+    receive do
+      {:last_message, 0, ^last_message_0} ->
+        IO.puts("Got last message from partition 0")
+    end
+
+    receive do
+      {:last_message, 1, ^last_message_1} ->
+        IO.puts("Got last message from partition 1")
+    end
+
+    # Let's wait to see if we get more messages
+    Process.sleep(1000)
+
+    messages = Agent.get(messages_agent, & &1)
+
+    on_exit(fn ->
+      stop_broadway(broadway_pid)
+    end)
+
+    {:ok, %{broadway_pid: broadway_pid, messages: messages}}
+  end
+
+  test "number of processed messages = total messages ", %{messages: messages} do
+    assert length(messages) == Config.n_messages()
+  end
+
+  test "messages are not duplicated", %{messages: messages} do
+    messages_with_count =
+      Enum.reduce(messages, %{}, fn msg, acc ->
+        Map.update(acc, msg.data, %{count: 1, list: [msg]}, fn %{count: count, list: list} ->
+          %{list: [msg | list], count: count + 1}
+        end)
+      end)
+
+    duplicated_messages = Enum.filter(messages_with_count, fn {_k, v} -> v.count > 1 end)
+
+    assert duplicated_messages == []
+  end
+
+  test "order of messages and offsets", %{messages: messages} do
+    assert get_ordering_proplems(messages) == []
+  end
+
+  defp reset_topic(topic) do
+    :os.cmd('kafka-topics --delete --zookeeper localhost:2181 --topic #{topic}')
```

Comment:
Use something like this so we get the output when running tests too?

    System.cmd "kafka-topics", ["--delete", "--zookeeper", "localhost:2181", "--topic", topic],
      into: IO.stream(:stdio, :line), stderr_to_stdout: true